File: Mandatory/002 - Repeated String/repeatedstring.py
import math

def repeatedString(evaluatedString, numberCharacters):
  arrayEvalChars = list(evaluatedString)
  
  repetitions = math.floor(numberCharacters / len(arrayEvalChars))
  
  count_a = repetitions * len(list(filter(lambda char : char == 'a', arrayEvalChars)))
                 
  surplus = numberCharacters % len(arrayEvalChars)

  if surplus > 0:
    arraySurplus = arrayEvalChars[0:surplus]    
    count_a +=  len(list(filter(lambda char : char == 'a', arraySurplus)))
  
  return count_a 

print(repeatedString('aba', 10))

File: Mandatory/002 - Repeated String/repeatedstring.js
function repeatedString(evaluatedString, numberCharacters) {
  const arrayEvalChars = evaluatedString.split('');

  const repetitions = Math.floor(numberCharacters / arrayEvalChars.length);

  let count_a = repetitions * arrayEvalChars.filter(char => char === 'a').length;

  let surplus = numberCharacters % arrayEvalChars.length;

  if (surplus > 0) {
    const arraySurplus = arrayEvalChars.slice(0, surplus);
    count_a += arraySurplus.filter(char => char === 'a').length;
  }

  return count_a;
}

console.log(repeatedString('aba', 10));
